base/inc/shortcode.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/**
+ * A simple shortcode that just renders a
+ *
+ * @param $attr
+ * @param string $content
+ *
+ * @return string
+ */
+function siteorigin_widget_shortcode( $attr, $content = '' ){
+	$attr = shortcode_atts( array(
+		'class' => false,
+		'id' => '',
+	), $attr, 'siteorigin_widget' );
+
+	global $wp_widget_factory;
+	if( ! empty( $attr[ 'class' ] ) && isset( $wp_widget_factory->widgets[ $attr[ 'class' ] ] ) ) {
+		$the_widget = $wp_widget_factory->widgets[ $attr[ 'class' ] ];
+
+		$meta = json_decode( htmlspecialchars_decode( $content ), true );
```

Comment:
Cool, the only reason I can think of not to use `$content` is if we expect to ever want to actually use it for content. :stuck_out_tongue: